lib/charms/mongodb/v0/mongodb_backups.py

```diff
@@ -46,26 +48,35 @@ def __init__(self, charm, substrate="k8s"):
         super().__init__(charm, "client-relations")
         self.charm = charm
         self.substrate = substrate
-        self.framework.observe(self.charm.on.config_changed, self._on_pbm_config_changed)
-
-    def _on_pbm_config_changed(self, event) -> None:
-        """Handles PBM configurations."""
-        # handling PBM configurations requires that the pbm snap is installed.
-        if "db_initialised" not in self.charm.app_peer_data:
-            logger.debug("Cannot set PBM configurations, MongoDB has not yet started.")
-            event.defer()
-            return
+        self.framework.observe(self.charm.on.create_backup_action, self._on_create_backup_action)
 
-        snap_cache = snap.SnapCache()
-        pbm_snap = snap_cache["percona-backup-mongodb"]
+    def _on_create_backup_action(self, event) -> None:
+        if self.model.get_relation(S3_RELATION) is None:
+            event.fail("Relation with s3-integrator charm missing, cannot create backup.")
+            return
 
-        if not pbm_snap.present:
-            logger.debug("Cannot set PBM configurations, PBM snap is not yet installed.")
-            event.defer()
+        try:
+            snap_cache = snap.SnapCache()
+            pbm_snap = snap_cache["percona-backup-mongodb"]
+            if not pbm_snap.present:
+                logger.debug("Cannot start PBM agent, PBM snap is not yet installed.")
+                event.defer()
+                return
+            pbm_snap.start(services=["pbm-agent"])
+            # sleep for 10 seconds while pbm snap service starts. Without this running backup will
+            # occasionally fail.
+            time.sleep(10)
+        except snap.SnapError as e:
+            logger.error("An exception occurred when starting pbm agent, error: %s.", str(e))
+            event.fail(f"Failed to backup MongoDB with error: {str(e)}")
             return
 
-        # TODO handle PBM configurations set by the user
-        self.charm.unit.status = ActiveStatus()
+        try:
+            subprocess.check_output("percona-backup-mongodb backup", shell=True)
```

Comment:
Don't you need to specify the `PBM_MONGODB_URI`  environment variable or the `--mongodb-uri` command line argument when running the backup?

Reply:
This is set when the s3-relation is joined (seperate PR)